src/App.tsx

```diff
@@ -1,56 +1,104 @@
-import React from 'react';
+import React, { useCallback, useMemo, useState } from 'react';
 import './App.scss';
+import classNames from 'classnames';
 import { peopleFromServer } from './data/people';
+import { Person } from './types/Person';
 
-export const App: React.FC = () => {
-  const { name, born, died } = peopleFromServer[0];
+const debounce = (
+  f: {
+    (value: React.SetStateAction<string>): void,
+    (...args: [string]): void,
+  }, delay: number | undefined,
+) => {
+  let timerId: NodeJS.Timeout;
+
+  return (...args: [string]) => {
+    clearTimeout(timerId);
+    timerId = setTimeout(f, delay, ...args);
+  };
+};
+
+export const App = () => {
+  const [query, setQuery] = useState('');
+  const [appliedQuery, setAppliedQuery] = useState('');
+  const [closeList, setCloseList] = useState(false);
+  const [selectedPerson, setSelectedPerson] = useState<Person | null>(null);
+
+  const applyQuery = useCallback(
+    debounce(setAppliedQuery, 1000),
+    [],
+  );
+
+  const visiblePeople = useMemo(() => {
+    return peopleFromServer.filter(
+      person => person.name.toLowerCase().includes(appliedQuery.toLowerCase()),
+    );
+  }, [appliedQuery]);
+
+  function handleSaveSelected(event: React.MouseEvent<HTMLDivElement>) {
+    const findPerson = peopleFromServer.find(
+      person => person.name === event.currentTarget.textContent,
+    ) || peopleFromServer[0];
+
+    setSelectedPerson(findPerson);
+    setCloseList(prevState => !prevState);
+  }
+
+  function handleInput(event: React.ChangeEvent<HTMLInputElement>) {
+    setQuery(event.target.value);
+    applyQuery(event.target.value);
+    setCloseList(false);
+  }
+
+  const error = visiblePeople.length === 0;
```

Comment:
```suggestion
  const error = !!visiblePeople.length;
```
and maybe not need addittional variable error and just check it in jsx